### PR TITLE
Gonzales 3.2 update

### PR DESCRIPTION
**Pull request by DanPurdy**
Updates develop with the latest version of Gonzales.

All tests should be passing but we need to just make sure there's no issues with installing on windows as there's a related issue over with gonzales about this currently.

`DCO 1.1 Signed-off-by: Dan Purdy danjpurdy@gmail.com`

### CODE REVIEW

**Comment by bgriffith** [comment]
It's passing! :tada: :beers:

**Comment by DanPurdy** [comment]
Windows tests for gonzales-pe install and then we move forward!

**Comment by DanPurdy** [comment]
Ok i think I'm just going to fork and release a version of gonzales-pe without the bash specific post install scripts and see if we can get this running on windows again.. It's the only barrier left..

File: lib/helpers.js [hunk 1]
```diff
@@ -4,16 +4,7 @@ var util = require('util'),
...
-
-/**
- * Easy access to the 'merge' library's cloning functionality
- * @param   {object} obj Object to clone
- * @returns {object}     Clone of obj
- */
-var clone = function (obj) {
-  return merge(true, obj);
-};
+    gonzales = require('gonzales-pe');
```
Comment (Snugug) [inline lib/helpers.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320542015/issues/source?bid=3169912&fileBranchId=3169916#l7)

File: lib/helpers.js [hunk 2]
```diff
@@ -407,7 +291,7 @@ helpers.attemptTraversal = function (node, traversalPath) {
  *                                       (without '.', '#', etc) resulting from suffix extensions
  */
 helpers.collectSuffixExtensions = function (ruleset, selectorType) {
-  var parentSelectors = helpers.attemptTraversal(ruleset, ['selector', 'simpleSelector', selectorType, 'ident']),
+  var parentSelectors = helpers.attemptTraversal(ruleset, ['selector', selectorType, 'ident']),
```
Comment (Snugug) [inline lib/helpers.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320542015/issues/source?bid=3169912&fileBranchId=3169916#l294)

File: lib/rules/extends-before-declarations.js [hunk 3]
```diff
@@ -13,25 +13,25 @@ module.exports = {
       var lastDeclaration = null;
 
       block.forEach(function (item, j) {
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Function 'anonymous' has a complexity of 6.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l15)

File: lib/rules/extends-before-declarations.js [hunk 4]
```diff
@@ -13,25 +13,25 @@ module.exports = {
...
-        if ((item.type === 'include' || item.type === 'extend') &&
+        // TODO: Remove tempory fix - atrule type is work around for issue:
+        // https://github.com/tonyganch/gonzales-pe/issues/147
+        if ((item.is('include') || item.is('extend') || item.is('atrule')) &&
             item.first('atkeyword')) {
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l19)

File: lib/rules/extends-before-declarations.js [hunk 5]
```diff
@@ -13,25 +13,25 @@ module.exports = {
...
-                  'column': item.start.column,
-                  'message': 'Extends should come before declarations',
-                  'severity': parser.severity
-                };
-                result = helpers.addUnique(result, error);
-              });
+              error = {
+                'ruleId': parser.rule.name,
+                'line': item.start.line,
+                'column': item.start.column,
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l25)

File: lib/rules/extends-before-declarations.js [hunk 6]
```diff
@@ -13,25 +13,25 @@ module.exports = {
...
-                  'ruleId': parser.rule.name,
-                  'line': item.start.line,
-                  'column': item.start.column,
-                  'message': 'Extends should come before declarations',
-                  'severity': parser.severity
-                };
-                result = helpers.addUnique(result, error);
-              });
+              error = {
+                'ruleId': parser.rule.name,
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l23)

File: lib/rules/extends-before-declarations.js [hunk 7]
```diff
@@ -13,25 +13,25 @@ module.exports = {
       var lastDeclaration = null;
 
       block.forEach(function (item, j) {
-        if ((item.type === 'include' || item.type === 'extend') &&
+        // TODO: Remove tempory fix - atrule type is work around for issue:
+        // https://github.com/tonyganch/gonzales-pe/issues/147
+        if ((item.is('include') || item.is('extend') || item.is('atrule')) &&
             item.first('atkeyword')) {
           if (item.first('atkeyword').first('ident').content === 'extend') {
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l20)

File: lib/rules/extends-before-declarations.js [hunk 8]
```diff
@@ -13,25 +13,25 @@ module.exports = {
...
-                  'message': 'Extends should come before declarations',
-                  'severity': parser.severity
-                };
-                result = helpers.addUnique(result, error);
-              });
+              error = {
+                'ruleId': parser.rule.name,
+                'line': item.start.line,
+                'column': item.start.column,
+                'message': 'Extends should come before declarations',
```
Comment (Snugug) [inline lib/rules/extends-before-declarations.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/snugug/sass-lint/file/2320582947/issues/source?bid=3169912&fileBranchId=3169916#l26)

**Comment by DanPurdy** [comment]
Go away codacy... 

@Snugug can you change the quotes parameter in codacy's code-settings to "single" please I don't have access to the parameters on codacy. You may be able to add us to a codacy organisation i think?

**Pull request merged**